### PR TITLE
Add Execution and Value Store bitesize docs and examples [BW-435]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Review note: 
---

May be best to review the docs in their rendered form in case there are markdown errors that aren't obvious in the MD files:

* https://cromwell.readthedocs.io/en/cjl_execution_store_docs/developers/bitesize/workflowExecution/executionStore/
* https://cromwell.readthedocs.io/en/cjl_execution_store_docs/developers/bitesize/workflowExecution/valueStore/
* https://cromwell.readthedocs.io/en/cjl_execution_store_docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples/

### CODE REVIEW

File: docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+|10|`Done`|`Done`|`Running`|`Running`|`NotStarted`|`NotStarted`|`NotStarted`
+|11|`Done`|`Done`|`Running`|`Done`|`NotStarted`|`NotStarted`|`NotStarted`
+|12|`Done`|`Done`|`Done`|`Done`|`NotStarted`|`NotStarted`|`NotStarted`
+
+As the results for each shard come in, the value store is also updated to include them:
+
+At step 11 (shard 1 has finished but shard 0 has not):
+```json
+{
+  "x": "[0, 1, 2]",
```
Comment (salonishah11) [inline docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md:170]: The value of `x` here and everywhere below should be `[0, 1]`

File: mkdocs.yml [hunk 2]
```diff
@@ -96,6 +96,10 @@ nav:
                 - Job Key / Value Store: developers/bitesize/workflowExecution/jobKeyValueStore.md
```
Comment (salonishah11) [inline mkdocs.yml:96]: Is `Job key value store` and `Value Store` the same thing? If yes, then we should remove this entry as it is simply a placeholder

File: docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md (hunk 1 above)
Comment (cjllanwarne) [reply docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md:170]: Thanks! I made the workflow simpler to produce less rows in the tables, but forgot to update this part 🤦‍♂️

File: mkdocs.yml (hunk 2 above)
Comment (cjllanwarne) [reply mkdocs.yml:96]: Unfortunately not. The Job KV store is a way for jobs to record (eg) PAPI ids in the database in a backend agnostic way.

Comment (salonishah11) [reply mkdocs.yml:96]: Oh I see. Well then please ignore my comment :)

**Review by salonishah11 — APPROVED** [review]
Thank you for adding the examples. It makes understanding both these stores easier 👍  💯

File: docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  "single_task.string_out": "hello",
+  "string_out": "hello"
+}
+``` 
+
+Cromwell can use this information to trigger the "workflow complete" logic.
+
+## Handling scatters
+
+When we run scattered tasks, the Execution Store cannot tell ahead of time how many
```
Comment (aednichols) [inline docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md:76]:
```suggestion
When Cromwell runs scattered tasks, the Execution Store cannot tell ahead of time how many
```
[ToL] I think a detached/impersonal voice works better for docs (here and elsewhere)

**Review by aednichols — APPROVED** [review]
👍

File: docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md (hunk 3 above)
Comment (cjllanwarne) [reply docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md:76]: Kate would be proud! I'll do my best to reword these

**Pull request merged**